src/os/ObjectStore.cc

```diff
@@ -66,6 +66,14 @@ ObjectStore *ObjectStore::create(CephContext *cct,
 				 const string& journal,
 				 osflagbits_t flags)
 {
+  if (type == "random" &&
```

Comment:
might want to guard the `if` block with `#if defined(HAVE_LIBAIO)`, otherwise the FreeBSD build is broken with this change. and since we always have libaio with our linux builds, it's safe to have this guard.